test/passes/inlining_all-features.wast

```diff
@@ -64,7 +64,7 @@
  (export "func_36_invoker" (func $1))
  (func $0
   (return_call_ref
-   (ref.null (func))
+   (ref.null $none_=>_none)
```

Comment:
Would it make sense to add this as a new case rather than replacing the slightly different `ref.nul (func)` case?

Reply:
At least based on [the issue you filed](https://github.com/WebAssembly/function-references/issues/42) it seemed like it wasn't intended to work? This PR removes the old form and just supports the form we all agree should work.

Reply:
Ok, sounds good to me in the context of GC. For the basic reference types feature, we will still need to support `(ref.null func)` and `(ref.null extern)`, but that can be rectified in a separate change. https://webassembly.github.io/reference-types/core/text/instructions.html#reference-instructions

Reply:
We do still support `(ref.null func)` - this just removes the "expanded inline" variant of `(ref.null (func))` (i.e. it disallows the extra parens around the `func`). That is tested in `test/reference-types.wast`.